packages/kbn-guided-onboarding/src/components/landing_page/guide_card_footer.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { EuiButton, EuiProgress, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { GuideId, GuideState } from '../../types';
+import { UseCase } from './use_case_card';
+
+export const GuideCardFooter = ({
+  guides,
+  useCase,
+  activateGuide,
+}: {
+  guides: GuideState[];
+  useCase: UseCase;
+  activateGuide: (useCase: UseCase, guideState?: GuideState) => void;
+}) => {
+  const guideState = guides.find((guide) => guide.guideId === (useCase as GuideId));
+  const viewGuideButton = (
+    <div className="eui-textCenter">
+      <EuiButton
+        // Used for FS tracking
+        data-test-subj={`onboarding--guideCard--view--${useCase}`}
+        fill
+        onClick={() => activateGuide(useCase, guideState)}
+      >
+        {i18n.translate('guidedOnboardingPackage.gettingStarted.guideCard.startGuide.buttonLabel', {
+          defaultMessage: 'View guide',
+        })}
+      </EuiButton>
+    </div>
+  );
+  // guide has not started yet
+  if (!guideState || guideState.status === 'not_started') {
+    return viewGuideButton;
+  }
+  const { status, steps } = guideState;
+  const numberSteps = steps.length;
+  const numberCompleteSteps = steps.filter((step) => step.status === 'complete').length;
+  const stepsLabel = i18n.translate('guidedOnboardingPackage.gettingStarted.guideCard.stepsLabel', {
+    defaultMessage: '{progress} steps',
+    values: {
+      progress: `${numberCompleteSteps}/${numberSteps}`,
+    },
+  });
+  // guide is completed
+  if (status === 'complete') {
+    return (
+      <>
+        <EuiProgress
+          valueText={stepsLabel}
+          value={numberCompleteSteps}
+          max={numberSteps}
+          size="s"
+          label={i18n.translate(
+            'guidedOnboardingPackage.gettingStarted.guideCard.progress.completedLabel',
+            {
+              defaultMessage: 'Completed',
+            }
+          )}
+        />
+        <EuiSpacer size="l" />
+        <div className="eui-textCenter">
```

Comment:
I followed your suggestion to not use a flexGroup when there is only 1 component. Please let me know if there is a better way to center the button in the footer.

Reply:
Sorry,  maybe my comment was clear initially. We shouldn't be using `eui-` classes directly in our React components, as they are internal to EUI and could possibly change. I meant - you can use `EuiFlexGroup` to achieve this (it's fine even for one component). Another way is to set the `div` width to 50% and  margin to `auto`:
https://jsfiddle.net/79kjtwqo/1/

Reply:
I don't think `eui-textCenter` is internal to EUI, or at least is is documented as an available utility class here - https://elastic.github.io/eui/#/theming/typography/utilities.

Reply:
I agree with Alison here, if the EUI class should not be used in Kibana, there should be an eslint rule for that.

Reply:
@majagrubic was absolutely correct here.

@alisonelizabeth it's not that it's internal to EUI, it's that they're likely not going to exist for much longer.  Even on the [documentation link](https://elastic.github.io/eui/#/theming/typography/utilities) you reference, it states:

>For most of your usages we recommend using [EuiText](https://elastic.github.io/eui/#/display/text) or [EuiTitle](https://elastic.github.io/eui/#/display/title) components directly instead of these theme tokens. Or head to the [text utilities page](https://elastic.github.io/eui/#/theming/typography/utilities) for helper classes and functions.

EUI is moving to Emotion:

> EUI is transitioning to a CSS-in-JS solution for theming and requires consuming/customizing global variables in both the Sass and CSS-in-JS methods. To track this effort, subscribe to the [Meta ticket](https://github.com/elastic/eui/issues/3912).

Using these classes directly should be for edge cases only.  The right solution for this would have been to use `EuiText`, rather than a `div` with a global EUI classname.

@yuliacech The reason there's no lint rule is that it would literally blow up the linter, due to the number of exceptions we've made in the past, (or older cases before Emotion and theming).

I would recommend a follow-up PR to switch to `EuiText`.

Reply:
> @alisonelizabeth it's not that it's internal to EUI, it's that they're likely not going to exist for much longer. 

👍 thanks for clarifying

> The right solution for this would have been to use EuiText, rather than a div with a global EUI classname.

This is for centering a button though, right @yuliacech? I don't _think_ `EuiText` is appropriate in that case. Perhaps @majagrubic earlier suggestion is the best approach:

> What I meant was - you can use EuiFlexGroup to achieve centering of the button (it's fine even for one component). Another way is to set the div width to 50% and margin to auto:
> https://jsfiddle.net/79kjtwqo/1/

I've opened https://github.com/elastic/kibana/issues/143794 to follow up on this.

Reply:
Thanks a lot for the context, @clintandrewhall! I just wish this were communicated more broadly because with the number of Kibana contributors we won't be able to address the issue via individual code reviews. An eslint rule might be a grind at the beginning to disable all lines with classes, but at least it would prevent new ones from being added. Otherwise a documentation update might be an easier alternative. From the text, it's not clear that the classes will be deleted soon and currently should not be used in production, maybe a deprecation could be added there?